test/common/stats/thread_local_store_test.cc

```diff
@@ -189,6 +260,152 @@ TEST_F(StatsThreadLocalStoreTest, BasicScope) {
   // Includes overflow stat.
   EXPECT_CALL(*this, free(_)).Times(5);
 }
+TEST_F(StatsThreadLocalStoreTest, BasicSingleHistogramMerge) {
+  InSequence s;
+  store_->initializeThreading(main_thread_dispatcher_, tls_);
+
+  Histogram& h1 = store_->histogram("h1");
+  EXPECT_EQ("h1", h1.name());
+
+  expectCallAndAccumulate(h1, 0);
+  expectCallAndAccumulate(h1, 43);
+  expectCallAndAccumulate(h1, 41);
+  expectCallAndAccumulate(h1, 415);
+  expectCallAndAccumulate(h1, 2201);
+  expectCallAndAccumulate(h1, 3201);
+  expectCallAndAccumulate(h1, 125);
+  expectCallAndAccumulate(h1, 13);
+
+  EXPECT_EQ(1, validateMerge());
+
+  store_->shutdownThreading();
+  tls_.shutdownThread();
+
+  // Includes overflow stat.
+  EXPECT_CALL(*this, free(_));
```

Comment:
Can you explain in more detail what's going on here?

Reply:
I am really not sure about this. This is followed in all other tests and with out this the test fails with assertions.So just added it.